[Nigel Tan] iP

nniiggeell opened the pull request:
# Lebron Chat Bot
> “You have to be able to accept failure to get better." -Lebron James

Lebron Chat Bot frees your mind of having to remember things you need to do. It's,
* Text-based
* Easy to learn
* Super fast to use!
All you need to do is:,
1. download it from https://github.com/nniiggeell/ip
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!
Features:
- [x]  Managing tasks
- [x] Managing deadlines 
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
    public static void main(String[] args) throws Exception {
          new Lebron().run();
      }
```

Commit:
Add Gradle support

nniiggeell committed:
Add Response.java

nniiggeell committed:
Update Duke.java for Level-1

nniiggeell committed:
Update .gitignore to ignore .class files

nniiggeell committed:
Update Response.java for level-2

nniiggeell committed:
Update Duke.java for Level-2

nniiggeell committed:
Update .gitignore

nniiggeell committed:
Add Task.java

nniiggeell committed:
Update Response.java for Level-3

nniiggeell committed:
Update Duke.java for level-4

nniiggeell committed:
Add javadocs to Response.java

nniiggeell committed:
Add javadocs to Task.java

nniiggeell committed:
Add javadocs to Duke.java

nniiggeell committed:
Update Task.java toString method

nniiggeell committed:
Add ToDo.java

nniiggeell committed:
Add Events.java

nniiggeell committed:
Add Deadline.java

nniiggeell committed:
Update Task.java change the toString

nniiggeell committed:
Update Response.java

nniiggeell committed:
Update ToDo.java

Remove pos from constructor

nniiggeell committed:
Update Task.java

Remove pos from constructor

nniiggeell committed:
Update Events.java

Remove pos from constructor

nniiggeell committed:
Update Deadline.java

Remove pos from constructor

nniiggeell committed:
Update Response.java

Update formatting of response

nniiggeell committed:
Update Duke.java

For level-4 requirements.
Added java enums.
Changed else if statements to case switch.

nniiggeell committed:
Update Duke.java

Fix formatting.

nniiggeell committed:
Update Response.java

Change the formatting of output.

nniiggeell committed:
Update text-ui-test

For A-TextUiTesting.

nniiggeell committed:
Update Duke.java

New Enum value for OTHER.
Handle error messages for Level-5

nniiggeell committed:
Update Response.java

Added delete method.

nniiggeell committed:
Update Response.java

Add javadocs

nniiggeell committed:
Update Duke.java

Added case for DELETE, for Level-6

nniiggeell committed:
Update Response.java

Changed the constant horizontalLine to HORIZONTAL_LINE

nniiggeell committed:
Update Duke.java

Reformat code with proper indents for switch case.

Change constant horizontalLine to HORIZONTAL_LINE

nniiggeell committed:
Add TaskTest.java

nniiggeell committed:
Add Storage.java

To handle the storage function for level-7.

nniiggeell committed:
Update Deadline.java

nniiggeell committed:
Update Duke.java

Provided run() function to handle the logic.

nniiggeell committed:
Update Events.java

nniiggeell committed:
Update Response.java

nniiggeell committed:
Update Task.java

nniiggeell committed:
Update ToDo.java

nniiggeell committed:
Update .gitignore

Ignore duke.txt

nniiggeell committed:
Update .gitignore

Ignore duke.txt

nniiggeell committed:
Update Deadline.java

To store LocalDate.

nniiggeell committed:
Update Events.java

To store LocalDate.

nniiggeell committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	.gitignore

Resolution:
Use the one from master branch.

nniiggeell committed:
Merge branch 'branch-Level-8'

nniiggeell committed:
Refactor Response.java to Ui.java

For A-MoreOOP, Ui.java now just handles the bot responding.

nniiggeell committed:
Update Deadline.java

nniiggeell committed:
Update Duke.java

For A-MoreOOP.

nniiggeell committed:
Update Events.java

For A-MoreOOP.

nniiggeell committed:
Update Storage.java

Fix formatting

nniiggeell committed:
Update Task.java

Fix formatting.

nniiggeell committed:
Update ToDo.java

Fix formatting.

nniiggeell committed:
Refactor Response.java to Ui.java

For A-MoreOOP.

nniiggeell committed:
Add Parser.java

For A-MoreOOP.

nniiggeell committed:
Add TaskList.java

For A-MoreOOP.

nniiggeell committed:
Update TaskTest.java

Included imports from packages.

nniiggeell committed:
Move files to new directory

Included imports from packages.
Included files to packages.

nniiggeell committed:
Move files to new directory

For A-Packages, included packages and import statements.

nniiggeell committed:
Rename Lebron.java

Rename from Duke.java to Lebron.java
Move file to new directory and included packages and imports.

nniiggeell committed:
Remove TaskTest/TaskTest.java

Moved elsewhere

nniiggeell committed:
Add DeadlineTest.java

JUnit Test.

nniiggeell committed:
Add TaskListTest.java

JUnit test.

nniiggeell committed:
Add TaskStub.java

Stub class for Task.java

nniiggeell committed:
Add Test/TaskTest.java

Moved the file to this location.

nniiggeell committed:
Add UiStub.java

A stub file for Ui.java

nniiggeell committed:
Fix javadoc for Lebron.java

nniiggeell committed:
Fix javadoc for Storage.java

nniiggeell committed:
Update Lebron.java

Add case for find event.

nniiggeell committed:
Update Ui.java

Add find event. For level-9

nniiggeell committed:
Update TaskList.java

Fix formatting

nniiggeell committed:
Update Lebron.java

Fix formatting

nniiggeell committed:
Update Storage.java

Fix formatting

nniiggeell committed:
Update Ui.java

Fix formatting

nniiggeell committed:
Update Events.java

Fix formatting

nniiggeell committed:
Merge branch 'branch-Level-9'

nniiggeell committed:
Merge branch 'branch-A-Javadoc'

nniiggeell committed:
Merge branch 'branch-A-CodingStandard'

nniiggeell committed:
Create jar file

nniiggeell committed:
Update formatting of Lebron.java and TaskList.java

nniiggeell committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2122S1/ip

nniiggeell committed:
Add gradle checkstyle

nniiggeell committed:
Move Test folder inside src

nniiggeell committed:
Move test files to src/test/java

nniiggeell committed:
Update build.gradle

nniiggeell committed:
Update Deadline.java

Fix format

nniiggeell committed:
Merge branch 'ip-week-4'

nniiggeell committed:
Update build.gradle

For javafx

nniiggeell committed:
Update Lebron.java

For level-10

nniiggeell committed:
Update Ui.java

For level-10

nniiggeell committed:
Update TaskList.java

For level-10

nniiggeell committed:
Update TaskList.java

For level-10

nniiggeell committed:
Update UiStub.java

For level-10

nniiggeell committed:
Add javafx files

For level-10 GUI

nniiggeell committed:
Merge branch 'branch-Level-10'

nniiggeell committed:
Update .gitignore

nniiggeell committed:
Update Launcher.java

Following tutorial part 4

nniiggeell committed:
Update Lebron.java

Following tutorial part 4, remove the start() method

nniiggeell committed:
Update Ui.java

Remove horizontal lines

nniiggeell committed:
Add fxml files and controller files

Following tutorial part 4

pranav-ganesh commented on src/main/java/lebron/task/TaskList.java:
```diff
@@ -0,0 +1,75 @@
+    /**
+     * Constructor.
+     *
+     * @param lst The list of tasks.
```
Shouldn't parameter lebron be included in the javadoc?

pranav-ganesh commented on src/main/java/lebron/task/TaskList.java:
```diff
@@ -0,0 +1,75 @@
+    /**
+     * This method handles the markDone event when the user says done, marking the Task as done.
+     *
+     * @param pos the position of the Task in the ArrayList lst
```
Shouldn't there be an @return statement in your javadoc since your method returns a value? I noticed the same issue in several other places too...

pranav-ganesh commented on src/main/java/lebron/task/Task.java:
```diff
@@ -0,0 +1,62 @@
+        return isDone ? "1" : "0";
+    }
+
+    public String toFile() {
```
Shouldn't method names be verbs or contain action words? Not a significant issue though! It would be more ideal if your method name started with a verb, like get, set.....

pranav-ganesh commented on src/main/java/lebron/Ui.java:
```diff
@@ -0,0 +1,109 @@
+ * @author Nigel Tan
+ */
+public class Ui {
+    private final String HORIZONTAL_LINE = "    ____________________________________________________________\n";
```
Line looks longer than 110 characters. Don't you think It would be ideal to split it into two lines to avoid very long lines?

nniiggeell replied:
Thanks for spotting this!

pranav-ganesh commented on src/main/java/lebron/Storage.java:
```diff
@@ -0,0 +1,111 @@
+                break;
+            default:
+                System.out.println(HORIZONTAL_LINE
+                        + "    :( OOPS! I'm sorry, but I don't know what that means.\n"
```
I don't see how splitting this print statement into 3 lines improves readability significantly when compared to 2 lines, do you? Wouldn't it be better to just split it into 2 lines since that won't violate the coding standard in any possible way?

pranav-ganesh commented on src/main/java/lebron/controller/MainWindow.java:
```diff
@@ -0,0 +1,70 @@
+package lebron.controller;
+
+import javafx.scene.control.Label;
```
Isn't this inconsistent ordering of import statements? In other files you have added spaces between different types of import statements, here you haven't. Again not a significant issue, but remember that a consistent ordering of import statements makes it easier to browse the list and determine the dependencies when there are many imports.

pranav-ganesh commented on src/main/java/lebron/task/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    @Override
+    public String toString() {
```
Shouldn't there be a javadoc for this toString method? Not sure if you have left it out purposely or not!

pranav-ganesh commented on src/main/java/lebron/task/Task.java:
```diff
@@ -0,0 +1,62 @@
+    /**
+     * This method marks the task as true.
+     */
+    public void mark() {
```
Isn't the method name 'mark' slightly more ambiguous than 'markAsDone' for example? It is ambiguous in this case whether mark denotes mark as true or mark as false (unless I read the javadoc or look at the method implementation).

pranav-ganesh commented on src/test/java/UiStub.java:
```diff
@@ -0,0 +1,22 @@
+import lebron.Lebron;
```
Shouldn't every class be part of some package? Refer to the Java coding standard (Package and Import Statements section) regarding this.

pranav-ganesh commented on src/main/java/lebron/Lebron.java:
```diff
@@ -0,0 +1,173 @@
+            String commandWord = parser.parseText(text);
+            String[] splitWords = parser.split(text);
+            Command command = Command.fromString(commandWord);
+            switch (command) {
```
I like the fact that you used a switch statement rather than if else! 😄

pranav-ganesh reviewed (commented):
Overall I think you have done a good job and your code is well written! Just a few nits to fix here and there!

nniiggeell committed:
Move DialogBox.java to controller package

nniiggeell committed:
Refactor test files to test package

nniiggeell committed:
Disable checkstyle in build.gradle

nniiggeell committed:
Fix formatting and codestyle

nniiggeell committed:
Update Storage.java

Use try catch blocks.

nniiggeell committed:
Add Exception classes

nniiggeell committed:
Update Lebron.java

nniiggeell committed:
Update Storage.java

nniiggeell committed:
Update Ui.java

Fix javadocs

nniiggeell committed:
Update MainWindow.java

nniiggeell committed:
Update Lebron.java

nniiggeell committed:
Update Storage.java

nniiggeell committed:
Update Ui.java

fix method name for coding standards

nniiggeell committed:
Update Deadline.java

To accept time

nniiggeell committed:
Update Events.java

To accept time.

nniiggeell committed:
Update DeadlineTest.java

nniiggeell committed:
Update Lebron.java

Lebron class does not utilise assertions to check the current program
state.

Assertions are included to let the program check the current program
state so that runtime can verify.

Adding assertions allow us to check if the program holds the assumptions
we expect.

Let's write some assertions.

Using assertions has benefits to check if the programmer ie. me has made
any mistakes!

nniiggeell committed:
Update TaskList.java and TaskListTest.java

To check if delete has appropriate index and throw relevant exception.

nniiggeell committed:
Update Storage, Ui, Deadline.java

Improve code quality.

nniiggeell committed:
Merge pull request #1 from nniiggeell/branch-A-Assertions

Add assertions

nniiggeell committed:
Merge branch 'branch-A-Assertions' of https://github.com/nniiggeell/ip into branch-A-Code-Quality

nniiggeell committed:
Merge pull request #2 from nniiggeell/branch-A-Code-Quality

Improve code quality

nniiggeell committed:
Merge branch 'master' of https://github.com/nniiggeell/ip

nniiggeell committed:
Fix code quality

Fix javadocs.

nniiggeell committed:
Update Lebron.java and TaskList.java

For C-Undo

nniiggeell committed:
Update files

nniiggeell committed:
Update Lebron.java

The run() method in Lebron.java has arrowhead style, due to the
long switch cases and long logic in each case.

Abstracting out these logic will make for cleaner and clearer code.

Let's extract those logic into separate methods so that there is SLAP.
Doing so will make the code easier to read and understand.

nniiggeell committed:
SLAP Storage.java

nniiggeell committed:
Update Ui.java

Fix javadocs

nniiggeell committed:
Update Deadline.java

Fix javadocs

nniiggeell committed:
Update Events.java

nniiggeell committed:
Update Task.java

nniiggeell committed:
Update TaskList.java

nniiggeell committed:
Merge pull request #4 from nniiggeell/branch-A-CodeQuality

Branch a code quality

nniiggeell committed:
Merge branch 'master' of https://github.com/nniiggeell/ip

nniiggeell committed:
Try multiple Undo

nniiggeell committed:
Fix Undo feature

nniiggeell committed:
Fix formatting and javadoc

nniiggeell committed:
Merge branch 'Trying-Undo'

nniiggeell committed:
Improve GUI

nniiggeell committed:
Update formatting

nniiggeell committed:
Update README.md

nniiggeell committed:
Set theme jekyll-theme-tactile

nniiggeell committed:
Set theme jekyll-theme-midnight

nniiggeell committed:
Add Ui.png

nniiggeell committed:
Merge branch 'master' of https://github.com/nniiggeell/ip

nniiggeell committed:
Delete InvalidInputException.java

Unused

nniiggeell committed:
Update README.md

nniiggeell committed:
Fix formatting and javadocs

nniiggeell committed:
Update README.md

nniiggeell committed:
Update README.md

nniiggeell committed:
Fix formatting
